RecoTracker/MkFitCore/src/PropagationMPlex.cc

```diff
@@ -589,15 +589,15 @@ namespace mkfit {
 
   //==============================================================================
 
-  void propagateHelixToZMPlex(const MPlexLS& inErr,
-                              const MPlexLV& inPar,
-                              const MPlexQI& inChg,
-                              const MPlexQF& msZ,
-                              MPlexLS& outErr,
-                              MPlexLV& outPar,
-                              const int N_proc,
-                              const PropagationFlags pflags,
-                              const MPlexQI* noMatEffPtr) {
+  void __attribute__((optimize("no-inline"))) propagateHelixToZMPlex(const MPlexLS& inErr,
```

Comment:
Please remove my tag from this thread

Reply:
@srlantz

Reply:
Yes, I found it was the combination of `-ffast-math` and inlining that seemed to be the source of gcc's woes in compiling `propagateHelixToZMPlex`. I chose to disable inlining because it's my understanding that `-ffast-math` is necessary to enable auto-vectorization of math functions in gcc. (Experiments on a very simple code showed that the specific required suboptions of -ffast-math are `-fno-math-errno -ffinite-math-only -fno-rounding-math -funsafe-math-optimizations`.)

However, perhaps switching to `no-fast-math` is indeed less detrimental to the performance of `propagateHelixToZMPlex`. If it turns out that the implementation of this function no longer vectorizes, then it may benefit from the same treatment that @gartung gave to helixAtR, i.e., breaking the body of the function into many smaller, vectorizable loops. That way only the trig functions would potentially be left as the serial slow spots.

Reply:
So I removed all of the attributes and ran under gdb. I turned the ternary operators in propagateHelixToZMPlex into if then else statements. I still get the segfault. The results of `hypot=std::hypot(outPar(1,0,0), outPar(1,0,0))` returns `nan` and `rbin=Config::materialEff.getRbin(hypot)` returns `-1`. Even though rbin is checked, `if (zbin >= 0 && zbin < Config::nBinsZME && rbin >= 0 && rbin < Config::nBinsRME)` `hitsRl(n, 0, 0) = Config::materialEff.getRlVal(zbin, rbin);` is still called.  

My guess is `-fno-math-errno` causes the issue.